crates/cli/commands/src/stage/dump/merkle.rs

```diff
@@ -139,9 +146,18 @@ fn unwind_and_copy<N: ProviderNodeTypes>(
 
     let unwind_inner_tx = provider.into_tx();
 
-    // TODO optimize we can actually just get the entries we need
-    output_db
-        .update(|tx| tx.import_dupsort::<tables::StorageChangeSets, _>(&unwind_inner_tx))??;
+    // Import only the StorageChangeSets entries within the block range (from, to)
+    output_db.update(|tx| {
+        let mut source_cursor = unwind_inner_tx.cursor_dup_read::<tables::StorageChangeSets>()?;
+        let mut target_cursor = tx.cursor_dup_write::<tables::StorageChangeSets>()?;
+
+        let range = BlockNumberAddress::range(from..=to);
+        for row in source_cursor.walk_range(range)? {
```

Comment:
@wetkeyboard were you able to test this code with a storage that has multiple keys? My understanding of `walk_range` is that it does not handle dup keys, it would only walk over the first subkey in every primary key, but I could be wrong.

Reply:
Hey @wetkeyboard sorry I think I steered you wrong here, walk_range also works on dup tables. You can see an example of iterating over `StorageChangeSets` [here](https://github.com/paradigmxyz/reth/blob/aa192c255b53335ba509987c82ab26e79e954195/crates/trie/db/src/state.rs#L233-L241)

Could you change it back to just use `walk_range`?